libcst/__init__.py

```diff
@@ -200,6 +200,18 @@
     BasicPositionProvider,
     SyntacticPositionProvider,
 )
+from libcst.metadata.scope_provider import (
+    Access,
+    Assignment,
```

Comment:
I'm worried about this being confused with Assign/AnnAssign/AugAssign nodes already present. Same with BaseAssignment below. Should we standardize instead on having people import from `libcst.metadata`? When matchers are added, they will not be included in `libcst/__init__.py`.

Reply:
Moved all the metadata providers and its data structure to namespace libcst.metadata to prevent confusion.